extensions/2.0/Khronos/KHR_behavior/README.md

```diff
@@ -62,6 +62,290 @@ From these requirements, the main features can be deducted:
+
+Each `function` is synchronous and a given `function` graph can be thought of as a call stack, beginning with an `action` parameter and calling each parameter `function` in turn up the stack until it terminates at either a constant or a query. The stacks are then unwound down to produce the final requested parameter `value`. 
+
+Execution is deterministic because the `functions` are all synchronous and pure, meaning they have no state and no side-effects. They cannot affect the scene states they are querying directly: only `action` nodes can do that. A `function` graph runs before the `action` that calls it via its parameter list, and after that `action`'s input `action` completed. This means if both of these actions are changing the same scene state value, that value will be deterministically in the final state of the input `action` when queried. 
+
+In addition to querying scene properties and animation states, the state of `passive` triggers can also be queried, which can be especially useful in combination with a gate `action`. 
+
+### Security requirements for Level 0
+
+Replacing constant parameters with `functions` is allowed in general across Level 0, as long as all functions are pure, stateless, and side-effect-free. Like `actions`, `functions` must only reference indices less than their own to prevent recursion. They do not impact the fundamental restriction that the Turing machine can only advance via explicit user activation.
+
+### Functions categorized
+
+`query`:
+- property
+- animationTime
+- animationSpeed
+- audioTime
+- audioPlaying
+- hover
+- proximity
+- inView
+
+`math`:
+- see list below
+
+`channel`:
+- see list below
+
+### Examples
+
+This example's first behavior opens the door when it is clicked, and closes it when clicked again. No need for two animations: it is simply played backward via the `speed` parameter. This also means it will operate naturally when clicked in rapid succession: opening partly, then closing again without discontinuity. 
+
+The second behavior indicates to the user they should try selecting the door by blinking it red when they get close to it and it is in view, or if they hover over it. This action will reoccur whenever the user moves away and returns. It uses `functions` to accomplish this by gating on the value of the e.g. proximity state. This is different than a `waitAll` because in that case the user would have to *both* move away and turn away, then come back before it would re-fire. Instead it will refire even when the user turns away and back again without moving.
+
+```json
+{
+  "functions":[
+    {// 0
+      "name": "Near Door",
+      "type": "query/proximity",
+      "parameters": {
+        "mesh": 5
+      }
+    },
+    {// 1
+      "name": "Looking at Door",
+      "type": "query/inView",
+      "parameters": {
+        "mesh": 5
+      }
+    }
+  ],
+  "actions":[
+    // Door activation behavior
+    {// 0
+      "name": "Activate Door",
+      "type": "active/select",
```

Comment:
I want to leave the category as part of the type simply because I want to make it easy to inspect that the Level 0 rules are followed. This way a verifier doesn't have to go back to the spec (and any extensions) to create of list of allowed types. I'm not attached to any of the names though.